lib/statsd.js

```diff
@@ -527,19 +470,7 @@ Client.prototype._close = function (callback) {
   }
 
   try {
-    if (this.protocol === TCP) {
-      this.socket.destroy();
-    } else if (this.protocol === UDS) {
-      this.socket.close();
-      // this close is synchronous, and the socket will not emit a
-      // close event, so we can callback right away
-      if (callback) {
-        callback();
-      }
-    }
-    else {
-      this.socket.close();
-    }
+    this.socket.close();
```

Comment:
How about also setting `this.socket = undefined`, so there's no doubt things are broken?

(And then it's trivial to do `if (!this.socket) { this.socket = createSocket(...) };` to attempt self-healing later on :)

Reply:
Well, I'm trying not to change the logic in this PR and leaving behaviour as is. I better leave it intact.

Another place that bugs me a lot is https://github.com/brightcove/hot-shots/pull/129/files#diff-6b52e4167496e1de6769211f454200f7R80 - basically swallowing errors sometimes. Wish we'd just threw it, at least at setup stage.